server/netty-server/src/main/scala/sttp/tapir/server/netty/NettySocketConfig.scala

```diff
@@ -0,0 +1,65 @@
+ *
+ * @param reuseAddress
+ *   Re-use address.The value of this socket option is a Boolean that represents whether the option is enabled or disabled. The exact semantics of this socket option are socket type and system dependent.
+ *   In the case of stream-oriented sockets, this socket option will usually determine whether the socket can be bound to a socket address when a previous connection involving that socket address is in the TIME_WAIT state. On implementations where the semantics differ, and the socket option is not required to be enabled in order to bind the socket when a previous connection is in this state, then the implementation may choose to ignore this option.
+ *   For datagram-oriented sockets the socket option is used to allow multiple programs bind to the same address. This option should be enabled when the socket is to be used for Internet Protocol (IP) multicasting.
+ *
+ * @param typeOfService
+ *   The Type of Service (ToS) octet in the Internet Protocol (IP) header.
+ *   The value of this socket option is an Integer representing the value of the ToS octet in IP packets sent by sockets to an IPv4 socket. The interpretation of the ToS octet is network specific and is not defined by this class. Further information on the ToS octet can be found in RFC 1349 and RFC 2474. The value of the socket option is a hint. An implementation may ignore the value, or ignore specific values.
+ *   The initial/default value of the TOS field in the ToS octet is implementation specific but will typically be 0. For datagram-oriented sockets the option may be configured at any time after the socket has been bound. The new value of the octet is used when sending subsequent datagrams. It is system dependent whether this option can be queried or changed prior to binding the socket.
+*/
+case class NettySocketConfig(
+  sendBuffer: Option[Int],
+  receiveBuffer: Option[Int],
+  tcpNoDelay: Boolean,
+  reuseAddress: Boolean,
+  typeOfService: Option[Int]
+) {
+
+  def withSendBuffer(i: Int) = copy(sendBuffer = Some(i))
+  def withNoSendBuffer = copy(sendBuffer = None)
+  def withReceiveBuffer(i: Int) = copy(receiveBuffer = Some(i))
+  def withNoReceiveBuffer = copy(receiveBuffer = None)
+  def withTcpNoDelay = copy(tcpNoDelay = true)
+
+  def withNoTcpNoDelay = copy(tcpNoDelay = false)
+  def withReuseAddress = copy(reuseAddress = true)
+
+  def withNoReuseAddress = copy(reuseAddress = false)
+  def withTypeOfService(i: Int) = copy(typeOfService = Some(i))
+  def withNoTypeOfService = copy(typeOfService = None)
+
+}
+
+object NettySocketConfig {
+
+  def default: NettySocketConfig = NettySocketConfig(
+    sendBuffer = None,
+    receiveBuffer = None,
+    tcpNoDelay = false,
+    reuseAddress = false,
+    typeOfService = None
```

Comment:
same question - are there some defaults in Netty, or in other web servers, that would be sensible to use here?